ICAFIX/scripts/icaDim.m

```diff
@@ -237,11 +239,13 @@
 
 
 Out.data=single(zeros(OrigdataSizeOne,OrigdataSizeTwo));
-if DEMDT~=-1
-    Out.data(OrigdataMask,:)=data_trend + (((u*diag(Out.EigSAdj)*v')'+repmat(mean_data_detrend_vn,size_data_detrend_vnOne,1)) .* repmat(Out.noise_unst_std,1,size_data_detrend_vnTwo));
-else
-    Out.data(OrigdataMask,:)=data_trend + (((u*diag(Out.EigSAdj)*v')') .* repmat(Out.noise_unst_std,1,size_data_detrend_vnTwo));
-end
+%if DEMDT~=-1
+%    Out.data(OrigdataMask,:)=data_trend + (((u*diag(Out.EigSAdj)*v')'+repmat(mean_data_detrend_vn,size_data_detrend_vnOne,1)) .* repmat(Out.noise_unst_std,1,size_data_detrend_vnTwo));
+%else
+    Out.data(OrigdataMask,:)=((u*diag(Out.EigSAdj)*v')'); clear v;
```

Comment:
Expanding the data during these expressions may use more memory than using a second variable.  If everything outside the mask in Out.data is zero, consider setting it to the empty matrix (clear doesn't work), then setting it to `zeros` only after you have the computation result.

Reply:
I'm not following, but feel free to make and test further edits.  I need to move on to something else now.

Reply:
Also, there aren't any zeros in the current test because we removed them in the parent script.

Reply:
In other words:
```
Out.data = [];
tempvar = ((u*diag(Out.EigSAdj)*v')'); clear v; %are these already single?
tempvar = tempvar .* repmat(Out.noise_unst_std,1,size_data_detrend_vnTwo); 
￼tempvar = data_trend + tempvar;
<clear anything else you don't need>
Out.data = zeros(<something>, <something>, 'single');
Out.data(OrigdataMask,:) = tempvar;
clear tempvar;
```